.circleci/scan_artifacts_codebuild.ts

```diff
@@ -0,0 +1,51 @@
+import * as execa from 'execa';
+import * as path from 'path';
+
+// This list contains a platform agnostic list of paths where artifacts are stored after each test.
+import { ARTIFACT_STORAGE_PATH_ALLOW_LIST_CODEBUILD } from '../scripts/artifact-storage-path-allow-list-codebuild';
+
+const ROOT_FOLDER_ABSOLUTE_PATH = process.env.CODEBUILD_SRC_DIR as string;
+
+export const hasMatchingContentInFolder = (patterns: string[], folder: string, excludeFolder = '{node_modules,.cache,.git}'): boolean => {
+  console.log('Scanning folder:', folder);
+  const patternParam = patterns.reduce<string[]>((acc, v) => [...acc, '-e', v], []);
+
+  try {
+    execa.sync('grep', ['-r', `--exclude-dir=${excludeFolder}`, ...patternParam, folder]);
```

Comment:
does execa.sync throw if exit code is non-zero?

Reply:
[yes](https://github.com/sindresorhus/execa#childprocessresult)

Reply:
if there is a match, grep does not throw & we return true to purge the artifacts. Otherwise, we look at return code error.

Reply:
btw this is not changed from circleci, just changed a few paths (ROOT_FOLDER path)